frontend/dockerfile/instructions/commands_runmount.go

```diff
@@ -151,6 +160,25 @@ func parseMount(value string) (*Mount, error) {
 		}
 
 		value := parts[1]
+		// possible env var, defer for later
```

Comment:
what case is this?

Reply:
`parseMount` is called twice - once when the command is initially parsed, and then again when dispatching the command. For the first call, we don't have a way to expand variables.

Reply:
Why do we need to check for the case? Can't we just check `expander==nil`, then expand the value if set, if `from` changes then show an error.

Reply:
Note that linter also complains about this.

Reply:
I'm not sure I understand. 
We check `expander == nil` and expand the value if set. Right now, if the `from` value has a `$` I return an informative error. Anyway, this code changed a bit so make sure you're looking at the most recent version

Reply:
What I'm saying is that I don't think we need to check if the value contains a variable. It's redundant (and possibly could diverge from expander implementation). We can just call expander and make decisions based on the return value.

Reply:
Gotcha. Will do that, but that means the failure will happen in the dispatch stage rather than the parsing stage

Reply:
That's fine. Don't forget to squash commits as well.

Reply:
I implemented this logic, but then I noticed it fails long before we get to `dispatch`. 
Basically in the case that `from` is a variable, a new stage is added and its base name is invalid.
See `frontend/dockerfile/dockerfile2llb/convert.go:233`.
How should we proceed?

Reply:
I don't get it. Vars are still not allowed in `from` and should produce an error. (Could actually add a test for that). But to check that from doesn't contain a variable you don't need special duplicate logic but can just compare the value of the expander.

Reply:
The problem is that this isn't supported by the current flow:
1) Initial parsing. In this stage we don't have the `expander`, so we can't check the diff as suggested.
2) Processing the build stages and validation. Here, if a user used a variable in `from`, an error will be produced because of the stages validation (basically, a build stage name can't be uppercase or contain a `$`).
3) Dispatching commands. Here we can compare the expanded value to the original one, but we won't even get here because of the previous stage.

I hope that makes sense :)

Reply:
Ok, leave the validation for `from` to get a clearer error message. The others don't need it and can rely only on the expander implementation.

Reply:
Done. 
It came out a bit verbose, but it was either that or extracting this whole block to a separate function with some weird inputs/outputs.
I also thought about using an anonymous function so I could use closures, but I wasn't sure if that would be in the spirit of Go.
LMK if you have a better idea of if this is good enough